src/components/calcite-picker/calcite-picker.tsx

```diff
@@ -0,0 +1,277 @@
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() pickerSelectionChange: EventEmitter;
+
+  @Event() pickerRowsDeleted: EventEmitter;
+
+  @Listen("rowToggled")
+  rowToggledHandler(event) {
+    const { row, selected, shiftPressed } = event.detail;
+    if (selected) {
+      if (this.multiple && shiftPressed && this.lastSelectedRow) {
+        const start = this.rows.indexOf(this.lastSelectedRow);
+        const end = this.rows.indexOf(row);
+        this.rows.slice(Math.min(start, end), Math.max(start, end)).forEach((currentRow) => {
+          currentRow.toggleAttribute("selected", true);
+          this.selectedValues.add(currentRow);
+        });
+      } else {
+        this.selectedValues.add(row);
+      }
+    } else {
+      this.selectedValues.delete(row);
+    }
+    if (!this.multiple && selected) {
+      this.rows.forEach((item) => {
+        if (item !== row) {
+          this.deselectRow(item);
+        }
+      });
+    }
+    this.lastSelectedRow = row;
+    this.pickerSelectionChange.emit(this.selectedValues);
+  }
+
+  @Listen("rowDeleted")
+  rowDeletedHandler(event) {
+    const { row } = event.detail;
+    row.toggleAttribute("hidden", true);
+    this.deletedRows.add(row);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  setupDragAndDrop(): void {
+    const sortGroups = [this.el, ...Array.from(this.el.querySelectorAll("calcite-picker-group"))];
+    sortGroups.forEach((sortGroup) => {
+      Sortable.create(sortGroup, {
+        group: "whateva",
+        handle: ".handle",
+        draggable: "calcite-picker-row"
+      });
+    });
+  }
+
+  deselectRow(item) {
+    item.toggleAttribute("selected", false);
+    this.selectedValues.delete(item);
+  }
+
+  startEdit() {
+    this.editing = true;
+  }
+
+  cancelDelete() {
+    this.deletedRows.forEach((row: HTMLCalcitePickerRowElement) => {
+      row.toggleAttribute("hidden", false);
+    });
+    this.deletedRows = new Set();
+    this.editing = false;
+  }
+
+  confirmDelete() {
+    let selectedChanged = false;
+    this.deletedRows.forEach((row: HTMLCalcitePickerRowElement) => {
+      if (this.selectedValues.has(row.value)) {
+        this.selectedValues.delete(row.value);
+        selectedChanged = true;
+      }
+      row.remove();
+    });
+    if (selectedChanged) {
+      this.pickerSelectionChange.emit(this.selectedValues);
+    }
+    this.deletedRows = new Set();
+    this.editing = false;
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async getSelectedRows() {
```

Comment:
Right now, the only way to get the selected rows is to listen to a change event on the picker. I think being able to get the selected rows without requiring a trigger to change is a nice convenience addition. If you guys think it will cause more harm than good, we can remove it.

Reply:
Can’t they just query the dom for rows?

Reply:
> Can’t they just query the dom for rows?

Not unless they want to have a code split for one type of data vs another. The component is retruning an object containing all selected rows keyed by value (might move to a map). querySelectorAll returns a nodeList.

I don't think it's a question of why it should be there. It's a question of - is there a good reason to remove it. I don't see any problems in keeping it.